MusicXML.scm

```diff
@@ -172,40 +176,37 @@
               #f
               (let ((art-pair (assq (string->symbol atype) the-map)))
                 (if art-pair (cdr art-pair) #f))))))
-    (define (writetag tag) (writeln "<~A/>" tag))
+    (define (writetag t) `(,t))
     (define (writearticulations art-types)
       (if art-types
           (let ((arts (filter identity (map (picker art-map) art-types)))
                 (orns (filter identity (map (picker orn-map) art-types)))
                 (onots (filter identity (map (picker onot-map) art-types))))
-            (begin
-              (if (not (null? arts))
-                  (begin
-                    (writeln "<articulations>")
-                    (map writetag arts)
-                    (writeln "</articulations>")))
-              (if (not (null? orns))
-                  (begin
-                    (writeln "<ornaments>")
-                    (map writetag orns)
-                    (writeln "</ornaments>")))
-              (map writetag onots)))))
+            ;;(ly:message "ARTS ~A" arts)
+            `(
+              ,(if (not (null? arts)) `(articulations ,(map writetag arts)) '())
+              ,(if (not (null? orns)) `(ornaments ,(map writetag orns)) '())
+              ,(map writetag onots)))
+          '()))
     (define (writeslurs stype num)
       (if (and num (> num 0))
-          (begin
-            (writeln "<slur number=\"~A\" type=\"~A\"/>" num stype)
-            (writeslurs stype (- num 1)))))
+          (cons
+            `(slur (@ (number ,num)
+                      (type ,stype)))
+            (writeslurs stype (- num 1)))
+          '()))
     (define (writenotations chord tuplet art-types slur-start slur-stop)
       (if (or (pair? tuplet) (and (not chord) (or art-types slur-start slur-stop)))
-          (begin
-            (writeln "<notations>")
-            (writetuplet tuplet)
-            (if (not chord)
-                (begin
-                  (writearticulations art-types)
-                  (writeslurs 'start slur-start)
-                  (writeslurs 'stop slur-stop)))
-            (writeln "</notations>"))))
+          `(notations
+            ,(writetuplet tuplet)
+            ;;(ly:message "WA ~A" (writearticulations art-types))
```

Comment:
I don't think un-commenting this line will work as expected because it's inside the quoted list?  In any case, it's clearer to put it before the quoted list.